### PR TITLE
added epochs and randomized start date of training

### PR DESCRIPTION
**Pull request by andrewyuan02**

### CODE REVIEW

File: agent.py [hunk 1]
```diff
@@ -103,12 +103,21 @@ def call(self, states):
                 a tensor that summarize the states tensor of shape [batch_sz * self.actor_H2]
         """
         # extract info from states
+        state = states[0]
```
Comment (zhouzypaul) [inline agent.py]: where is this used?

File: experiment.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from preprocess import get_data
```
Comment (zhouzypaul) [inline experiment.py]: seems like we don't need this script. could this also be git ignored?

File: stock_env.py [hunk 3]
```diff
@@ -1,15 +1,16 @@
 import tensorflow as tf
 import numpy as np
 from preprocess import get_data
-
+from random import randint
 
 class StockEnv():
     def __init__(self,
+                 data,
```
Comment (zhouzypaul) [inline stock_env.py:9]: is this the output of preprocessing? could you update the doc string to specify?

File: stock_env.py [hunk 4]
```diff
@@ -39,8 +40,8 @@ def __init__(self,
         self.borrow_interest = borrow_interest_annual / days_per_year  # daily penalty on borrowed stocks
         self.transaction_penalty = transaction_penalty
 
-        data = get_data()  # pricing data
-        self.pricing_data = data[1] if is_testing else data[0]
```
Comment (zhouzypaul) [inline stock_env.py:43]: why is the `is_testing` arg being taken out? I thought this was some kind of train/test split

File: stock_env.py [hunk 5]
```diff
@@ -71,8 +72,10 @@ def generate_episode(self, model):
         past_num = model.past_num
         num_stocks = model.num_stocks
 
-        timestep = past_num  # we start on day number <past_num>
-        timestep_stop = tf.shape(self.pricing_data)[1] + 1 if self.max_days is None else timestep + self.max_days
+        initial_timestep = past_num
+        timestep = initial_timestep
+
+        timestep_stop = tf.shape(self.pricing_data)[1] + 1 # if self.max_days is None else timestep + self.max_days
```
Comment (zhouzypaul) [inline stock_env.py:81]: why is `self.max_days` being taken out

**Review by zhouzypaul — COMMENTED** [review]
Great work on the visualization! I have a few questions on `stock_env.py`, but @AdamWang00 should probably be the judge on those.

File: experiment.py (hunk 2 above)
Comment (andrewyuan02) [reply experiment.py]: okay

File: stock_env.py (hunk 3 above)
Comment (andrewyuan02) [reply stock_env.py:9]: okay

File: stock_env.py (hunk 5 above)
Comment (andrewyuan02) [reply stock_env.py:81]: haven't gotten it working yet

**Pull request merged**